pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -214,297 +184,69 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
 	return newIngressName
 }
 
-// newApplicationIngressControllerCR creates a new IngressController CR
-func newApplicationIngressControllerCR(ingressControllerCRName, scope, dnsName string, certificate *corev1.LocalObjectReference, matchLabels map[string]string) (*operatorv1.IngressController, error) {
+func generateIngressController(appIngress v1alpha1.ApplicationIngress) *operatorv1.IngressController {
 	loadBalancerScope := operatorv1.LoadBalancerScope("")
-	switch scope {
+	switch appIngress.Listening {
 	case "internal":
 		loadBalancerScope = operatorv1.InternalLoadBalancer
 	case "external":
 		loadBalancerScope = operatorv1.ExternalLoadBalancer
 	default:
-		return &operatorv1.IngressController{}, errors.New("ErrCreatingIngressController")
+		loadBalancerScope = operatorv1.ExternalLoadBalancer
 	}
 
 	return &operatorv1.IngressController{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      ingressControllerCRName,
+			Name:      getIngressName(appIngress.DNSName),
 			Namespace: ingressControllerNamespace,
 			Annotations: map[string]string{
 				"Owner": "cloud-ingress-operator",
 			},
 		},
 		Spec: operatorv1.IngressControllerSpec{
-			DefaultCertificate: certificate,
-			Domain:             dnsName,
+			DefaultCertificate: &corev1.LocalObjectReference{
+				Name: appIngress.Certificate.Name,
+			},
+			Domain: appIngress.DNSName,
 			EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
 				Type: operatorv1.LoadBalancerServiceStrategyType,
 				LoadBalancer: &operatorv1.LoadBalancerStrategy{
 					Scope: loadBalancerScope,
 				},
 			},
 			RouteSelector: &metav1.LabelSelector{
-				MatchLabels: matchLabels,
+				MatchLabels: appIngress.RouteSelector.MatchLabels,
 			},
 		},
-	}, nil
+	}
 }
 
-// convertIngressControllerToMap takes in on cluster ingresscontroller list and returns them as a map with key Spec.Domain and value operatorv1.IngressController
-func convertIngressControllerToMap(existingIngress []operatorv1.IngressController) map[string]operatorv1.IngressController {
-	ingressMap := make(map[string]operatorv1.IngressController)
+func checkDefaultIngressControllerStatus(ingressController operatorv1.IngressController, desiredSpec operatorv1.IngressControllerSpec) bool {
 
-	for _, ingress := range existingIngress {
-		ingressMap[ingress.Spec.Domain] = ingress
+	if !(desiredSpec.Domain == ingressController.Status.Domain) {
+		return false
 	}
-	return ingressMap
-}
-
-// checkExistingIngress returns false if applicationIngress do not match any existing ingresscontroller on cluster
-func checkExistingIngress(existingMap map[string]operatorv1.IngressController, publishingStrategyIngress *cloudingressv1alpha1.ApplicationIngress) bool {
-	if _, ok := existingMap[publishingStrategyIngress.DNSName]; !ok {
-		log.Info(fmt.Sprintf("IngressController for %q not found", publishingStrategyIngress.DNSName))
+	if !(desiredSpec.EndpointPublishingStrategy == ingressController.Status.EndpointPublishingStrategy) {
```

Comment:
can we assume `ingressController.Status.EndpointPublishingStrategy` always exist?

Reply:
I'd also check for capitalization here (ie internal vs Internal)

Reply:
This is only checked for the default ingresscontroller if the info is not in the spec. AFAIK the default IngressController CR will always have it in one or the other.

Reply:
Also, this is checking the generated IngresController, so both of these are the same types, either `ExternalLoadBalancer` or `InternalLoadBalancer` not the appInress directly.

Reply:
ah ok so its when we are checking the spec that we would need to worry about capitalization then :+1: